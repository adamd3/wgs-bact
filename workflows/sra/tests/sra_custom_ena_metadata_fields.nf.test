nextflow_workflow {

    name "Test workflow: sra/main.nf"
    script "../main.nf"
    workflow "SRA"
    tag "workflows"
    tag "workflows_sra"
    tag "multiqc_mappings_config"
    tag "sra_fastq_ftp"
    tag "sra_ids_to_runinfo"
    tag "sra_merge_samplesheet"
    tag "sra_runinfo_to_ftp"
    tag "sra_to_samplesheet"
    tag "sra_custom_ena_metadata_fields"

    test("Parameters: --nf_core_pipeline rnaseq --ena_metadata_fields ... --sample_mapping_fields ...") {

        when {
            workflow {
                """
                input[0] = Channel.from('ERR1160846', 'GSE214215', 'SRR12848126')
                """
            }
            params {
                outdir = "results"
                nf_core_pipeline = "rnaseq"
                ena_metadata_fields = "run_accession,experiment_accession,library_layout,fastq_ftp,fastq_md5"
                sample_mapping_fields = "run_accession,library_layout"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
