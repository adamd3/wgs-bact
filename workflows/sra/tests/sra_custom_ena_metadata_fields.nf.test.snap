{
    "Parameters: --nf_core_pipeline rnaseq --ena_metadata_fields ... --sample_mapping_fields ...": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "samplesheet.csv:md5,a9e9da506288c364af14d46b86dafeb1"
                ],
                "2": [
                    "id_mappings.csv:md5,8ed2bd72d432eff4a8ed7d909af6f60c"
                ],
                "3": [
                    [
                        "multiqc_config.yml:md5,3bc981f0de28023083cdf13691d249d5",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "4": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47",
                    "versions.yml:md5,e977678bd13a54fc9c31addf135be15e"
                ],
                "fastq": [
                    
                ],
                "mappings": [
                    "id_mappings.csv:md5,8ed2bd72d432eff4a8ed7d909af6f60c"
                ],
                "sample_mappings": [
                    [
                        "multiqc_config.yml:md5,3bc981f0de28023083cdf13691d249d5",
                        "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9"
                    ]
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,a9e9da506288c364af14d46b86dafeb1"
                ],
                "versions": [
                    "versions.yml:md5,1496d1cbc9041e07ab8a0c25f0b054d9",
                    "versions.yml:md5,45e4a369df2d3af023b8622aa3f6d5a8",
                    "versions.yml:md5,9b17045ca8bdc272cb3f9d349a81d206",
                    "versions.yml:md5,b21338b74b9a8fe0c7a114f7686c07cd",
                    "versions.yml:md5,b52279f7d6b891a6523d9321f3f85b47",
                    "versions.yml:md5,e977678bd13a54fc9c31addf135be15e"
                ]
            }
        ],
        "timestamp": "2023-10-18T09:36:57.154883885"
    }
}